libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -418,7 +418,7 @@ bool ExpressionCompiler::visit(BinaryOperation const& _binaryOperation)
 		{
 			return dynamic_cast<Literal const*>(&_e) || _e.annotation().type->category() == Type::Category::RationalNumber;
 		};
-		bool swap = m_optimize && TokenTraits::isCommutativeOp(c_op) && isLiteral(rightExpression) && !isLiteral(leftExpression);
+		bool swap = m_optimiseOrderLiterals && TokenTraits::isCommutativeOp(c_op) && isLiteral(rightExpression) && !isLiteral(leftExpression);
```

Comment:
There are a lot of changes in this single PR.  I'd prefer to split the Yul part out, but that might be a lot of unnecessary work.

Can this single change split out however? This one is simple enough and is reasonable on its own.

Reply:
What exactly do you mean by this change? Just the renaming?

Reply:
Ignore this for the moment, will review commit by commit.